### PR TITLE
Additional features: limit number of comments and deletion

### PR DESCRIPTION
**Pull request by tatiana-s**
The comment section now supports:

- displaying a limited number of results where the number can be selected by the user, and
- deleting all comments at once is now possible.

(To see changes, a dev server can be run using `mvn package appengine:run` in `step/portfolio/`)

### CODE REVIEW

File: portfolio/src/main/webapp/style.css [hunk 1]
```diff
@@ -72,12 +72,39 @@ input[type=submit] {
...
 .comment {
   border-bottom: 1px solid white;
   margin-bottom: 10px;
   padding: 5px;
 }
 
+#number {
+  background-color: yellowgreen;
+  color: white;
+  border: none;
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/style.css:87]: (nit) please move border above color :)

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -61,3 +63,10 @@ function createCommentElement(text) {
   element.className = "comment";
   return element;
 }
+
+/** Deletes all comments on by the server.*/
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/script.js]: (nit) add space after dot.

File: portfolio/src/main/webapp/style.css [hunk 3]
```diff
@@ -72,12 +72,39 @@ input[type=submit] {
...
+  margin: 5px;
+  max-width: 100%;
+  padding: 5px 20px;
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/style.css]: I think 20px padding is too much - it makes the number disappear for me. The problem might be that it is in the middle of the menu but is hidden by the dropdown arrow.

File: portfolio/src/main/java/com/google/sps/data/CommentEntity.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * Specifies the kind and property names to use for comment entities in the datastore databse. 
+ */
+public enum CommentEntity {
+  KIND("Comment"),
+  CONTENT_PROPERTY("content"),
+  AUTHOR_PROPERTY("author"),
+  TIME_PROPERTY("time"),
+  MOOD_PROPERTY("mood");
+
+  private final String label;
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/data/CommentEntity.java:28]: It's not completely clear to me what label is - maybe use a better name (or at the very least add a comment:))

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 5]
```diff
@@ -31,34 +31,37 @@
 
 /** Servlet for handling comments data */
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:34]: (nit) dot at the end of a sentence

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 6]
```diff
@@ -31,34 +31,37 @@
...
+public class CommentServlet extends HttpServlet {
 
-  /* Datastore data is represented by entities which have a kind and certain properties,
-   * and the constants below define the kind and property names for comment entities.
-   * Changing them would mean comments saved with a previous definition would not be displayed anymore. */
-  private static final String COMMENT_KIND = "Comment";
-  private static final String CONTENT_PROPERTY = "content";
-
-  /* This is the name of the input form which is defined in the html of the main page */
+  /* The name of the input form which is defined in the html of the main page. */
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: Comments should only be used in rare cases when you cannot make your code self-explanatory. I think in this case you could just rename the constant to INPUT_FORM_NAME and remove the comment.  

Please think about whether the comments below are needed or adjusting the name will be enough.

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 7]
```diff
@@ -31,34 +31,37 @@
...
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query(COMMENT_KIND);
+    int commentNumber = Integer.parseInt(request.getParameter(COMMENT_NUMBER_PARAM));
+    Query query = new Query(CommentEntity.KIND.getLabel());
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
     List<String> comments = new ArrayList<>();
+    int count = 0;
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java]: You can use FetchOptions to limit the number of comments, see https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java [hunk 8]
```diff
@@ -31,34 +31,37 @@
...
   private static final String REDIRECT_URL = "/index.html";
 
+  /* The query string parameter for the number of comments to be displayed as defined in the script. */
+  private static final String COMMENT_NUMBER_PARAM = "number";
+
   /**
    * Loads and returns comments from the datastore database.
    */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:46]: I've tried adding comments and saw that order of the comments seems to be random. Is there a way to sort them by when they were added or is it not supported / should be implemented separately?

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+import com.google.sps.data.CommentEntity;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet for deleting comments */
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java]: (nit) dot

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+  /**
+   * Deletes all comments in the datastore database. 
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query(CommentEntity.KIND.getLabel());
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List<String> comments = new ArrayList<>();
```
Comment (nataliakrylova) [inline portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:45]: I don't see this list being used anywhere.

File: portfolio/src/main/webapp/index.html [hunk 11]
```diff
@@ -37,6 +37,14 @@ <h2>Comments</h2>
           <input type="submit"/>
         </form>
       <div id="text-container"></div>
+      <label for="comment-number">Maximum number of comments to display on one page: </label>
+      <select name="comment-number" id="number" onchange="showComments()">
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/index.html]: The ID is not very specific - could you please choose a name which indicates better what this is?

File: portfolio/src/main/webapp/script.js [hunk 12]
```diff
@@ -45,7 +45,9 @@ function showSlides(newIndex) {
 
 /** Displays list of comments returned by the server.*/
 function showComments() {
-  fetch('/comments').then(response => response.json()).then(comments => {
+  const numberForm = document.getElementById("number");
```
Comment (nataliakrylova) [inline portfolio/src/main/webapp/script.js]: I think this and the following lines can be merged into one :)

File: portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java (hunk 8 above)
Comment (tatiana-s) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:46]: It is not supported yet and sorting them by timestamp for example would be one of the features I am going to implement next.

Comment (nataliakrylova) [reply portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java:46]: Sounds great! :)

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java (hunk 10 above)
Comment (nataliakrylova) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:45]: Sorry my comment probably wasn't specific enough: I meant that comments list is not really used so I think you should remove it :)

**Review by nataliakrylova — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java (hunk 10 above)
Comment (tatiana-s) [reply portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java:45]: It is used though as far as I can tell ? Comments from the results get added to it and then it is passed in the response, I am not sure how to do that without the list.

**Pull request merged**